model/TestCenterService.php

```diff
@@ -201,4 +204,58 @@ public function getTestCenter($testCenterUri)
     {
         return new core_kernel_classes_Resource($testCenterUri);
     }
+
+    /**
+     * @param core_kernel_classes_Resource $testCenter
+     * @param User $user
+     * @param core_kernel_classes_Resource $role
+     * @return bool
+     * @throws TestCenterException
+     */
+    public function assignUser(\core_kernel_classes_Resource $testCenter, User $user, \core_kernel_classes_Resource $role)
+    {
+        $userResource = $this->getResource($user->getIdentifier());
+        $userRoles = $userResource->getPropertyValues($this->getProperty(UserRdf::PROPERTY_ROLES));
+
+        if (!in_array($role->getUri(), $userRoles)) {
```

Comment:
Is it required to check those roles which are not allowed? It could be the `default` case of the switch below.

Reply:
Here we check that user which is going to be assigned has appropriate role.
in the `switch` we check that test center supports given role. That are two different operations.